.github/workflows/winget.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish to Winget
+
+on:
+  release:
+    types: [published]
+
+env:
+  REGEX: 'wsl\.([\d.]+)\.x64\.msi$'
+
+jobs:
+  publish:
+    if: github.event.release.prerelease == false
+    runs-on: windows-latest # Action can only run on Windows
+    steps:
+      - name: Publish WSL
+        run: |
+          $assets = '${{ toJSON(github.event.release.assets) }}' | ConvertFrom-Json
+          $wingetRelevantAssetx64 = $assets | Where-Object { $_.name -like '*x64.msi' } | Select-Object -First 1
+          $wingetRelevantAssetARM64 = $assets | Where-Object { $_.name -like '*arm64.msi' } | Select-Object -First 1
+          $regex = [Regex]::New($env:REGEX)
+          $version = $regex.Match($wingetRelevantAssetx64.name).Groups[1].Value
```

Comment:
To simplify things you should be able to get the release version directly from `github.event.release.tag_name`